DOCSP-43091 Add What's New (#36)
mongodb/docs-c: source/whats-new.txt

.. _c-whats-new:

==========
What's New
==========

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: update, new feature, deprecation, upgrade

Learn what's new in:

* :ref:`Version 1.27.6 <c-version-1.27.6>`
* :ref:`Version 1.27.5 <c-version-1.27.5>`
* :ref:`Version 1.27.4 <c-version-1.27.4>`
* :ref:`Version 1.27.3 <c-version-1.27.3>`
* :ref:`Version 1.27.2 <c-version-1.27.2>`
* :ref:`Version 1.27.1 <c-version-1.27.1>`
* :ref:`Version 1.27.0 <c-version-1.27.0>`
* :ref:`Version 1.26.2 <c-version-1.26.2>`
* :ref:`Version 1.26.1 <c-version-1.26.1>`
* :ref:`Version 1.26.0 <c-version-1.26.0>`

.. _c-upcoming-breaking-changes:

.. _c-version-1.27.6:

What's New in 1.27.6
--------------------

The 1.27.6 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.6 release of ``libbson`` includes the following changes:

> 

Reviewer: S: since you use a combination of "Changes," "Fixes," "New Features," and "Improvements" subheaders, I'd make sure each bullet point without a subheader is categorized as one of those four things (although I'm not sure "Changes" is necessary). In this case, you could add the "Fixes" subheader:
```suggestion
Fixes
`````
```

Reviewer: Applies to all

- Fixes handling of malformed extended JSON for special BSON types.

libmongoc
~~~~~~~~~

The 1.27.6 release of ``libmongoc`` includes the following changes:

Fixes
`````

- Fixes TSan warning.
- Fixes C23 compile.

Improvements
```````````

- Documents expected behavior of command errors in a transaction.

.. _c-version-1.27.5:

What's New in 1.27.5
--------------------

The 1.27.5 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.5 release of ``libbson`` includes the following changes:

> - Fixes large string handling in ``bson_append_utf8``.

Reviewer: S: remove the period, since this isn't a complete sentence.
```suggestion
- Fixes large string handling in ``bson_append_utf8``
```

Reviewer: Applies to all (except bullet points that have two sentences, like lines 257-259).

libmongoc
~~~~~~~~~

The 1.27.5 release of ``libmongoc`` includes the following changes:

- Fixes possible build error in environments where ``bool`` or ``_Bool`` is a macro.

.. _c-version-1.27.4:

What's New in 1.27.4
--------------------

The 1.27.4 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.4 release of ``libbson`` includes the following changes:

- Fixes build issues with Visual Studio 2013.

libmongoc
~~~~~~~~~

The 1.27.4 release of ``libmongoc`` includes the following changes:

- Fixes possible crash reading malformed wired protocol message.
- Fixes build issues with Visual Studio 2013.

.. _c-version-1.27.3:

What's New in 1.27.3
--------------------

The 1.27.3 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.3 release of ``libbson`` includes the following changes:

Changes
```````

> - This patch introduces changes surrounding the development processes, 

Reviewer: S: reword slightly to match the other bullet points
```suggestion
- Introduces changes surrounding the development processes, 
```

~~~~~~~~~

The 1.27.3 release of ``libmongoc`` includes the following changes:

Fixes
`````

- Upgrades bundled Zlib from 1.2.3 to 1.3.1.

.. _c-version-1.27.2:

What's New in 1.27.2
--------------------

The 1.27.2 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.2 release of ``libbson`` includes the following changes:

Fixes
`````

- Fixes parsing of colon in timezone.
- Fixes error on failure to initialize a subdocument in ``bson_iter_visit_all``.

libmongoc
~~~~~~~~~

The 1.27.2 release of ``libmongoc`` includes the following changes:

Fixes
`````

- Closes lefotver connections to remote servers.

New Features
````````````

- Checks environment variable ``MONGO_EXPERIMENTAL_SRV_PREFER_TCP`` to prefer TCP over SRV lookup.

.. _c-version-1.27.1:

What's New in 1.27.1
--------------------

The 1.27.1 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

The 1.27.1 release of ``libbson`` includes the following changes:

- Fixes large string handling in ``bson_string_new`` and ``bson_string_append``.

libmongoc
~~~~~~~~~

> No changes since 1.27.0. Version incremented to match the ``libbson`` version.

Reviewer: S: make this a complete sentence
```suggestion
The version is incremented to match the ``libbson`` version, but there are no changes since 1.27.0.
```

Reviewer: Applies to all

.. _c-version-1.27.0:

What's New in 1.27.0
--------------------

The 1.27.0 release of ``libbson`` and ``libmongoc``, the libraries constituting 
the C driver, includes several changes.

libbson
~~~~~~~

> No changes since 1.26.2. Version incremented to match the ``libbson`` version.

Reviewer: S: reword to make these complete sentences
```suggestion
The version is incremented to match the ``libmongoc`` version, but there are no changes since 1.26.2.
```

Reviewer: Applies to all
